custom_components/tuya_local/devices/plantsio_ivy_smart_planter.yaml

```diff
@@ -0,0 +1,431 @@
+    name: PlantsIO Ivy Smart Planter
+primary_entity:
+  entity: sensor
+  name: Water level
+  dps:
+    - id: 108
+      type: integer
+      name: sensor
+      unit: mL
+      class: measurement
+      mapping:
+        - dps_val: 0
+          icon: "mdi:cup-outline"
+        - icon: "mdi:cup"
+    - id: 107
+      type: string
+      name: version
+secondary_entities:
+  # Sensor entities
+  - entity: sensor
+    name: Ambient light
+    class: illuminance
+    dps:
+      - id: 103
+        type: integer
+        name: sensor
+        unit: lx
+        class: measurement
+  - entity: sensor
+    name: Temperature
+    class: temperature
+    dps:
+      - id: 104
+        type: integer
+        name: sensor
+        unit: C
+        class: measurement
+        mapping:
+          - scale: 10
+  - entity: sensor
+    name: Humidity
+    class: humidity
+    dps:
+      - id: 105
+        type: integer
+        name: sensor
+        unit: "%"
+        class: measurement
+  - entity: sensor
+    name: Soil moisture
+    class: moisture
+    dps:
+      - id: 122
+        name: sensor
+        type: integer
+        optional: true
+        unit: "%"
+  - entity: sensor
+    name: Water status
+    icon: "mdi:watering-can"
+    class: enum
+    dps:
+      - id: 106
+        type: bitfield
+        name: sensor
+        mapping:
+          - dps_val: 0
+            value: "Initializing"
+          - dps_val: 2
+            value: "Drinking"
+          - value: "Unknown state"
+  - entity: sensor
+    name: Light status
+    icon: "mdi:brightness-5"
+    class: enum
+    dps:
+      - id: 138
+        type: bitfield
+        name: sensor
+        mapping:
+          - dps_val: 1
+            value: "Acceptable"
+          - dps_val: 2
+            value: "Good"
+          - dps_val: 3
+            value: "Exceptional"
+          - dps_val: 4
+            value: "Too Much"
+          - dps_val: 5
+            value: "Insufficient"
+          - value: "Unknown state"
+  - entity: sensor
+    name: Temperature status
+    icon: "mdi:thermometer"
+    class: enum
+    dps:
+      - id: 139
+        type: bitfield
+        name: sensor
+        mapping:
+          - dps_val: 1
+            value: "Good"
+          - dps_val: 2
+            value: "Hot"
+            icon: "mdi:thermometer-high"
+          - dps_val: 3
+            value: "Cold"
+            icon: "mdi:thermometer-low"
+          - value: "Unknown state"
+  - entity: sensor
+    name: Humidity status
+    class: enum
+    icon: "mdi:water-percent"
+    dps:
+      - id: 140
+        type: bitfield
+        name: sensor
+        mapping:
+          - dps_val: 1
+            value: "Good"
+          - value: "Unknown state"
+  # Diagnostic entities
+  - entity: sensor
+    name: Battery
+    class: battery
+    category: diagnostic
+    dps:
+      - id: 121
+        type: integer
+        name: sensor
+        unit: "%"
+  - entity: sensor
+    name: Charge state
```

Comment:
This can definitely be a binary_sensor (`class: battery_charging`), since the original value is boolean even